src/mixins/validatable-http-message.js

```diff
@@ -0,0 +1,450 @@
+
+    if (
+      this.validation.headers.realType ===
+        this.validation.headers.expectedType &&
+      expectsJson
+    ) {
+      return (this.validation.headers.validator = 'HeadersJsonExample');
+    } else {
+      this.validation.headers.validator = null;
+
+      if (!this.validation.headers.results) {
+        this.validation.headers.results = [];
+      }
+
+      const entry = {
+        message: `\
+No validator found for real data media type \
+"${JSON.stringify(this.validation.headers.realType)}" \
+and expected data media type \
+"${JSON.stringify(this.validation.headers.expectedType)}".\
+        `,
+        severity: 'error'
+      };
+
+      this.validation.headers.results.push(entry);
+    }
+  }
+
+  runHeadersValidator() {
+    // throw new Error JSON.stringify @validation.headers.validator, null, 2
+    let validator;
+
+    if (this.validation.headers.validator == null) {
+      this.validation.headers.rawData = null;
+    } else {
+      const ValidatorClass = validators[this.validation.headers.validator];
+      validator = new ValidatorClass(this.headers, this.expected.headers);
+      this.validation.headers.rawData = validator.validate();
+    }
+
+    if (!Array.isArray(this.validation.headers.results)) {
+      this.validation.headers.results = [];
+    }
+
+    if (this.validation.headers.rawData !== null) {
+      const results = validator.evaluateOutputToResults();
+      this.validation.headers.results = results.concat(
+        this.validation.headers.results
+      );
+    }
+  }
+
+  validateBody() {
+    this.validation.body = {};
+    this.validation.body.results = [];
+
+    this.setBodyRealType();
+    this.setBodyExpectedType();
+    this.setBodyValidator();
+    this.runBodyValidator();
+  }
+
+  setBodyRealType() {
+    this.validation.body.realType = null;
+    const bodyType = typeof this.body;
+
+    if (bodyType !== 'string') {
+      throw new Error(
+        `Expected HTTP body to be a String, but got: ${bodyType}`
+      );
+    }
+
+    let error;
+    const contentType = this.headers && this.headers['content-type'];
+
+    if (this.isJsonContentType(contentType)) {
+      try {
+        jph.parse(this.body);
+        this.validation.body.realType = contentType;
+      } catch (error1) {
+        error = error1;
+        const message = {
+          message: `\
+Real body 'Content-Type' header is '${contentType}' \
+but body is not a parseable JSON:\n${error.message}\
+`,
+          severity: 'error'
+        };
+
+        this.validation.body.results.push(message);
+      }
+    } else {
+      try {
+        jph.parse(this.body);
+        this.validation.body.realType = 'application/json';
+      } catch (error2) {
+        error = error2;
+        this.validation.body.realType = 'text/plain';
+      }
+    }
+  }
+
+  setBodyExpectedType() {
+    let error, message;
+    this.validation.body.expectedType = null;
+
+    if (this.validation.body.results == null) {
```

Comment:
Perhaps, but we must be careful about this. The way CoffeeScript worked is that it compared to `null` explicitly. I found quite a few places where it's based on that. Replacing with `!a` simply fails, as it will produce false negatives for things like empty strings or `false`. I will check this particular line, and remove null comparison if it doesn't break anything.